test/extended/util/wait.go

```diff
@@ -0,0 +1,41 @@
+package util
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+)
+
+func WaitForCMState(ctx context.Context, client corev1client.CoreV1Interface, namespace string, name string, condition func(cm *corev1.ConfigMap) (bool, error)) (*corev1.ConfigMap, error) {
```

Comment:
go doc

Reply:
Especialls that it falls over deletion

Reply:
will add it in a follow up to have this in place for the rebase and avoid going through CI flakes

Deletion isn't a state but an edge, it may be deleted before you start looking or never created at all. Also you get modified currently with deltaFIFO on recreate so it should use watch based Until, to see the edge, not informer based function.

But it fails on unexpected event so you know right away and won't merge a flake.